crates/ruff_linter/src/checkers/ast/analyze/expression.rs

```diff
@@ -96,6 +97,9 @@ pub(crate) fn expression(expr: &Expr, checker: &mut Checker) {
                     if checker.enabled(Rule::UnnecessaryTypeUnion) {
                         flake8_pyi::rules::unnecessary_type_union(checker, expr);
                     }
+                    if checker.enabled(Rule::RedundantNumericUnion) {
+                        flake8_pyi::rules::redundant_numeric_union(checker, expr);
+                    }
```

Comment:
Oh nice find. I think we should move this change out into its own PR so that it gets its own entry in the changelog. I have to think about it if it warrants a breaking change or not.

Reply:
I'll move the last commit into it's own PR